Add toastr.

lanieerts opened the pull request:
일단 성공 했을 때와 실패 했을 때로만 나눠 놓았습니다. 필요에 따라서 toastr.js에 추가하면 됩니다.

lanieerts committed:
Seperate reducer about comment and modify query.

lanieerts committed:
Modify detail of query for readMoreComments

lanieerts committed:
Modify url of action for createComment.

lanieerts committed:
Merge remote-tracking branch 'upstream/master'

pjhjohn committed:
Rebase commit of upstream/master.
Introduce react-router-bootstrap to avoid page refresh

Removed legacy style for Tab. Removed navbar radius by overriding style

Resolved #32 : Implemented Sticky Footer

Reference : https://getbootstrap.com/examples/sticky-footer

Disabled default visibility of ReduxDevTools

CRLF -> LF

reference : http://stackoverflow.com/questions/1889559/git-diff-to-ignore-m

Comment, PostDetail, Post re-styled with react-bootstrap

* Enabled well class in _bootstrap.scss

UI Enhancement using Panel Component

lanieerts committed:
Merge remote-tracking branch 'upstream/master'

lanieerts committed:
Merge remote-tracking branch 'upstream/master'

lanieerts committed:
Merge branch 'master' of https://github.com/lanieerts/jsonplaceholder-client

lanieerts committed:
Merge remote-tracking branch 'upstream/master'

lanieerts committed:
Merge remote-tracking branch 'upstream/master'

lanieerts committed:
Merge remote-tracking branch 'upstream/master'

lanieerts committed:
Merge remote-tracking branch 'upstream/master'

lanieerts committed:
Merge remote-tracking branch 'upstream/master'

lanieerts committed:
Merge remote-tracking branch 'upstream/master'

lanieerts committed:
Merge remote-tracking branch 'upstream/master'

lanieerts committed:
Merge remote-tracking branch 'upstream/master'

lanieerts committed:
Merge remote-tracking branch 'upstream/master'

lanieerts committed:
Add toastr.
When editing or writing post, toastr called.

pjhjohn commented on src/actions/index.js:
```diff
@@ -4,3 +4,4 @@ export * from './about';
 export * from './album';
 export * from './photo';
 export * from './readme';
+export * from './toastr';
```
toastr은 라이브러리의 이름일 뿐이고, 우리고 보통 사용하는 의미는 `notification`의 기능을 하는 `toast message`입니다.
따라서, `toastr` => `toast`로 바꾸는 것을 제안합니다.

lanieerts replied:
넵

pjhjohn commented on src/actions/toastr.js:
```diff
@@ -0,0 +1,10 @@
+export const CALL_TOASTR = 'CALL_TOASTR';
```
함수를 부른다는 뜻의 `CALL`인 건가요??
함수의 이름은 좀 더 의미를 담았으면 좋겠습니다. 간단히 `TOAST`만 있어도 될 것 같긴 하네요.
만약 여러 종류의 TOAST를 띄우는거라면 `TOAST_SOMETHINGMORE`가 되면 될 것 같구요

pjhjohn commented on src/actions/toastr.js:
```diff
@@ -0,0 +1,10 @@
+export const CALL_TOASTR = 'CALL_TOASTR';
+
+export function callToastr(action, status) {
```
우린 `Notification`을 주는거니까, 함수이름은 `Toast`도 필요없이 `notify`여도 충분할 것 같긴 하네요.
위의 것은 `NOTIFICATION`이 되어도 될 듯.

lanieerts replied:
그럼 callToastr => notify / CALL_TOASTR => NOTIFICATION으로 바꾸겠습니다.

pjhjohn commented on src/containers/Posts/PostNew.js:
```diff
@@ -16,7 +16,9 @@ class PostNew extends React.Component {
     this.setState({ 'disabled' : true });
-    this.props.createPost(props).then(() => {
+    this.props.createPost(props).then((response) => {
+      console.log(response);
```
`console.log`는 정말 필요한 경우가 아니라면 뺍시다.

lanieerts replied:
앗... ㅇ>ㅡ<

pjhjohn commented on src/containers/Toastr/Toastr.js:
```diff
@@ -0,0 +1,51 @@
+class Toastr extends React.Component {
+  static defaultProps = {
+    action: ``,
+    status: 404
```
`nothing` 의 `state`가 있어야할 것 같군요

lanieerts replied:
nothing의 state를 만든다면 어디에 사용되나요?

pjhjohn replied:
뭐....아무것도 안하면 되겠죠??
지금은 사실 404가 기본이면 뭔가를 하게 되지 않을까요?

lanieerts replied:
swtich case 문의 기준이 status가 아닌 action이라 따로 무언가를 하진 않습니다.
그리고 componentWillMoun에 넣어놓은거라 props 값이 변하지 않는 이상 아무것도 하지 않습니다.

pjhjohn replied:
그건 그렇긴 한데, 저번에 말씀드렸듯 `defaultProps` `initialState`에 들어가는 것 들은 [FSM](https://en.wikipedia.org/wiki/Finite-state_machine) 의 하나의 노드를 가리키게 되는게 좋습니다. 지금 우린 action: `` 과 status: 404 를 어디에서도 사용하고 있지 않으므로, default state로는 좀 부적합한 것 같다는 생각입니다.

lanieerts replied:
음.. 그럼 action의 default를 nothing으로, status의 default를 200으로 해두면 될까요.

pjhjohn commented on src/containers/Toastr/Toastr.js:
```diff
@@ -0,0 +1,51 @@
+  componentWillUpdate(nextProps) {
+    console.log(nextProps);
+    switch (nextProps.action) {
+      case `posting`: {
```
case는 action의 type처럼 const 필드를 만들어 사용합시다.

pjhjohn commented on src/containers/Toastr/Toastr.js:
```diff
@@ -0,0 +1,51 @@
+    console.log(nextProps);
+    switch (nextProps.action) {
+      case `posting`: {
+        if (nextProps.status === 201) this.refs.container.success(`Success! :)`, `Posting Result`);
```
포스팅 success는 201이 오나요?
이거 잘 생각해봐야 할 것 같은게, 200번대는 다 success라..

lanieerts replied:
Write Post가 DB에 정보를 하나 더 만드는거라 201이 옵니다. 이건 제가 response를 찍어서 확인 해 봤습니다.

lanieerts committed:
Modify issues of #61
1. Rename toastr => toast
2. Rename action => type & status => httpStatus
3. Rename callToastr => notify / CALL_TOASTR => NOTIFICATION
4. Remove consolo.log function.
5. Define some const for type in Toast.js.

lanieerts committed:
Modify form about export default in PostEdit.js
